sdk/notificationhubs/notification-hubs/review/notification-hubs.api.md

```diff
@@ -0,0 +1,177 @@
+    platform: "adm";
+}
+
+// @public
+export interface AppleInstallation extends DeviceTokenInstallation {
+    platform: "apple";
+}
+
+// @public
+export interface AppleMessage extends JSONNotificationMessage {
+    platform: "apple";
+}
+
+// @public
+export interface BaiduInstallation extends DeviceTokenInstallation {
+    platform: "baidu";
+}
+
+// @public
+export interface BaiduMessage extends JSONNotificationMessage {
+    platform: "baidu";
+}
+
+// @public
+export interface BrowserInstallation extends Installation {
+    platform: "browser";
+    pushChannel: BrowserPushChannel;
+}
+
+// @public
+export interface BrowserMessage extends JSONNotificationMessage {
+    platform: "browser";
+}
+
+// @public
+export interface BrowserPushChannel {
+    auth: string;
+    endpoint: string;
+    p256dh: string;
+}
+
+// @public
+export interface DeviceTokenInstallation extends Installation {
+    pushChannel: string;
+}
+
+// @public
+export interface FirebaseLegacyInstallation extends DeviceTokenInstallation {
+    platform: "gcm";
+}
+
+// @public
+export interface FirebaseLegacyMessage extends JSONNotificationMessage {
+    platform: "gcm";
+}
+
+// @public
+export interface Installation {
+    readonly expirationTime: string;
+    installationId: string;
+    readonly lastUpdate: string;
+    platform: string;
+    tags?: string[];
+    templates?: Record<string, InstallationTemplate>;
+    userId?: string;
+}
+
+// @public
+export interface InstallationPatch {
+    op: JSONPatchType;
+    path: string;
+    value?: string;
+}
+
+// @public
+export interface InstallationTemplate {
+    body: string;
+    expiry?: string;
+    headers: Record<string, string>;
+    tags?: string[];
+}
+
+// @public
+export type InstallationType = AppleInstallation | ADMInstallation | BaiduInstallation | BrowserInstallation | FirebaseLegacyInstallation | WindowsInstallation;
+
+// @public
+export interface JSONNotificationMessage extends NotificationHubMessage {
+    contentType: "application/json;charset=utf-8";
+}
+
+// @public
+export type JSONPatchType = "add" | "remove" | "replace";
+
+// @public
+export interface NotificationHubMessage {
+    body: string;
+    contentType: string;
+    headers: Record<string, string>;
+    platform: string;
+}
+
+// @public
+export type NotificationHubMessageType = AppleMessage | ADMMessage | BaiduMessage | BrowserMessage | FirebaseLegacyMessage | WindowsMessage | TemplateMessage;
+
+// @public
+export interface NotificationHubResponse {
+    correlationId?: string;
+    location?: string;
+    trackingId?: string;
+}
+
+// @public
+export class NotificationHubsClient extends ServiceClient {
+    constructor(connectionString: string, hubName: string, options?: NotificationHubsClientOptions);
+    deleteInstallation(installationId: string, options?: OperationOptions): Promise<NotificationHubResponse>;
+    getFeedbackContainerURL(options?: OperationOptions): Promise<URL>;
```

Comment:
any value in having this be a URL and not just a string?

Reply:
Not really, other than in .NET and other languages, we give it as a URL

Reply:
We stick to string for JS typically, since that's more idiomatic to real use and `URL` was only node global recently.